src/scripts/main.js

```diff
@@ -1,3 +1,22 @@
 'use strict';
 
-// write code here
+const allHeadlines = document.querySelectorAll('th');
+const tBody = document.querySelector('tbody');
+const allRows = Array.from(tBody.querySelectorAll('tr'));
+
+for (let i = 0; i < allHeadlines.length; i++) {
+  allHeadlines[i].addEventListener('click', () => {
+    const sortedRows = allRows.sort((a, b) => {
+      const aText = a.cells[i].textContent.trim();
+      const bText = b.cells[i].textContent.trim();
+
+      const aV = parseFloat(aText.replace(/[$,]/g, '')) || aText;
+      const bV = parseFloat(bText.replace(/[$,]/g, '')) || bText;
+
+      return aV < bV ? -1 : aV > bV ? 1 : 0;
```

Comment:
The sorting logic currently does not handle case-insensitive text sorting. If the table contains text data, consider converting both `aText` and `bText` to lowercase (or uppercase) before comparing to ensure consistent sorting regardless of text case.